src/main/java/ui/UI.java

```diff
@@ -268,7 +268,10 @@ private boolean isCloseOnQuit() {
         return commandLineArgs.getOrDefault("closeonquit", "false").equalsIgnoreCase("true");
     }
 
-    public void quit() {
+    public void quit(boolean isLogout) {
```

Comment:
quit will still have boolean to denote logout until UI level. Is this fine? Or should logout be called by other component?